### PR TITLE
Increase verbosity of "Premature end of potential file" error message

### PR DESCRIPTION
**Pull request by p-bro**
This pull request improves the usefulness of the error message of running out of data on reading tabulated potential files (format 3 and 4). 

I may have botched up the pull request commit messages.

### CODE REVIEW

File: src/potential_input_f3.c [hunk 1]
```diff
@@ -214,7 +220,8 @@ void read_pot_table3(char const* potential_filename, FILE* pfile,
        i < g_calc.paircol + 3 * g_param.ntypes; i++) {
     if (pstate->have_gradient) {
       if (2 > fscanf(pfile, "%lf %lf\n", val, val + 1))
-        error(1, "Premature end of potential file %s\n", pstate->filename);
+        error(1, "Premature end of potential file %s\n(no ADP-d grad, pot %d)\n",
```
Comment (dschopf) [inline src/potential_input_f3.c:223]:
```suggestion
        error(1, "Premature end of potential file %s\n(no s-band rho grad, pot %d)\n",
```

File: src/potential_input_f3.c [hunk 2]
```diff
@@ -231,7 +238,8 @@ void read_pot_table3(char const* potential_filename, FILE* pfile,
     /* read values */
     for (int j = 0; j < nvals[i]; j++) {
       if (1 > fscanf(pfile, "%lf\n", val))
-        error(1, "Premature end of potential file %s\n", pstate->filename);
+        error(1, "Premature end of potential file %s\n(no ADP-d values, pot %d line %d)\n",
```
Comment (dschopf) [inline src/potential_input_f3.c:241]:
```suggestion
        error(1, "Premature end of potential file %s\n(no s-band rho values, pot %d line %d)\n",
```

File: src/potential_input_f3.c [hunk 3]
```diff
@@ -247,7 +255,8 @@ void read_pot_table3(char const* potential_filename, FILE* pfile,
        i < g_calc.paircol + 4 * g_param.ntypes; i++) {
     if (pstate->have_gradient) {
       if (2 > fscanf(pfile, "%lf %lf\n", val, val + 1))
-        error(1, "Premature end of potential file %s\n", pstate->filename);
+        error(1, "Premature end of potential file %s\n(no ADP-q grad, pot %d)\n",
```
Comment (dschopf) [inline src/potential_input_f3.c:258]:
```suggestion
        error(1, "Premature end of potential file %s\n(no s-band F grad, pot %d)\n",
```

**Review by dschopf — CHANGES_REQUESTED** [review]
Changes look good except for the duplicate ADP lines.

**Comment by p-bro** [comment]
Oops. Cut and paste error.

**Review by dschopf — APPROVED** [review]

**Pull request merged**